ompi/mca/osc/ucx/osc_ucx_active_target.c

```diff
@@ -0,0 +1,264 @@
+            ucp_rkey_h rkey = (module->state_info_array)[ranks_in_win_grp[i]].rkey;
+            ucp_ep_h ep = OSC_UCX_GET_EP(module->comm, ranks_in_win_grp[i]);
+            uint64_t result_value;
+            void *req = ucp_atomic_fetch_nb(ep, UCP_ATOMIC_FETCH_OP_FADD, 1, &result_value,
+                                            sizeof(uint64_t), remote_addr, rkey,
+                                            amo_null_completion);
+            if (req != NULL) {
+                ucp_request_release(req);
+            }
+        }
+
+        free(ranks_in_grp);
+        free(ranks_in_win_grp);
+        ompi_group_free(&win_group);
+
+        OBJ_RELEASE(module->start_group);
+        module->start_group = NULL;
+
+        module->state.post_count = 0;
+    }
+
+    return ret;
+}
+
+int ompi_osc_ucx_post(struct ompi_group_t *group, int assert, struct ompi_win_t *win) {
+    ompi_osc_ucx_module_t *module = (ompi_osc_ucx_module_t*) win->w_osc_module;
+    int ret = OMPI_SUCCESS;
+
+    if (module->epoch_type.exposure != NONE_EPOCH) {
+        return OMPI_ERR_RMA_SYNC;
+    }
+
+    module->epoch_type.exposure = POST_WAIT_EPOCH;
+
+    if ((assert & MPI_MODE_NOCHECK) == 0) {
+        int i, size;
+        ompi_group_t *win_group = NULL;
+        int *ranks_in_grp = NULL, *ranks_in_win_grp = NULL;
+
+        OBJ_RETAIN(group);
+        module->post_group = group;
+
+        size = ompi_group_size(module->post_group);
+        ranks_in_grp = malloc(sizeof(int) * size);
+        ranks_in_win_grp = malloc(sizeof(int) * ompi_comm_size(module->comm));
+
+        for (i = 0; i < size; i++) {
+            ranks_in_grp[i] = i;
+        }
+
+        ret = ompi_comm_group(module->comm, &win_group);
+        if (ret != OMPI_SUCCESS) {
+            return OMPI_ERROR;
+        }
+
+        ret = ompi_group_translate_ranks(module->post_group, size, ranks_in_grp,
+                                         win_group, ranks_in_win_grp);
+        if (ret != OMPI_SUCCESS) {
+            return OMPI_ERROR;
+        }
+
+        for (i = 0; i < size; i++) {
+            uint64_t remote_addr = (module->state_info_array)[ranks_in_win_grp[i]].addr + sizeof(uint64_t); /* write to state.post_count on remote side */
+            ucp_rkey_h rkey = (module->state_info_array)[ranks_in_win_grp[i]].rkey;
+            ucp_ep_h ep = OSC_UCX_GET_EP(module->comm, ranks_in_win_grp[i]);
+            uint64_t result_value;
+            void *req = ucp_atomic_fetch_nb(ep, UCP_ATOMIC_FETCH_OP_FADD, 1, &result_value,
+                                            sizeof(uint64_t), remote_addr, rkey,
+                                            amo_null_completion);
+            if (req != NULL) {
+                ucp_request_release(req);
+            }
+        }
+
+        free(ranks_in_grp);
+        free(ranks_in_win_grp);
+        ompi_group_free(&win_group);
+
+        module->state.complete_count = 0;
+    } else {
+        module->post_group = NULL;
```

Comment:
I fixed as POST/START does not assign NULL pointer when NO_CHECK is passed, and WAIT/TEST always doing the synchronization. WAIT is at L228 and TEST is at L251.